openwisp_monitoring/db/backends/influxdb/client.py

```diff
@@ -0,0 +1,286 @@
+        retention_policies = db.get_list_retention_policies()
+        exists = False
+        duration_changed = False
+        for policy in retention_policies:
+            if policy['name'] == name:
+                exists = True
+                duration_changed = policy['duration']
+                break
+        if not exists:
+            db.create_retention_policy(name=name, duration=duration, replication=1)
+        elif exists and duration_changed:
+            db.alter_retention_policy(name=name, duration=duration)
+
+    def query(self, query, precision=None, **kwargs):
+        db = self.get_db()
+        database = kwargs.get('database') or self.db_name
+        return db.query(
+            query,
+            kwargs.get('params'),
+            epoch=precision,
+            expected_response_code=kwargs.get('expected_response_code') or 200,
+            database=database,
+        )
+
+    def write(self, name, values, **kwargs):
+        point = {
+            'measurement': name,
+            'tags': kwargs.get('tags'),
+            'fields': values,
+        }
+        timestamp = kwargs.get('timestamp')
+        if isinstance(timestamp, datetime):
+            timestamp = timestamp.strftime('%Y-%m-%dT%H:%M:%SZ')
+        if timestamp:
+            point['time'] = timestamp
+        self.get_db().write(
+            {'points': [point]},
+            {
+                'db': kwargs.get('database') or self.db_name,
+                'rp': kwargs.get('retention_policy'),
+            },
+        )
+
+    def read(self, key, fields, tags, **kwargs):
+        extra_fields = kwargs.get('extra_fields')
+        since = kwargs.get('since')
+        order = kwargs.get('order')
+        limit = kwargs.get('limit')
+        if extra_fields and extra_fields != '*':
+            fields = ', '.join([fields] + extra_fields)
+        elif extra_fields == '*':
+            fields = '*'
+        q = f'SELECT {fields} FROM {key}'
+        conditions = []
+        if since:
+            conditions.append(f'time >= {since}')
+        if tags:
+            conditions.append(
+                ' AND '.join(["{0} = '{1}'".format(*tag) for tag in tags.items()])
+            )
+        if conditions:
+            conditions = 'WHERE %s' % ' AND '.join(conditions)
+            q = f'{q} {conditions}'
+        if order:
+            q = f'{q} ORDER BY {order}'
+        if limit:
+            q = f'{q} LIMIT {limit}'
+        return list(self.query(q, precision='s').get_points())
+
+    def get_list_query(self, query, precision='s'):
+        return list(self.query(query, precision=precision).get_points())
+
+    def get_list_retention_policies(self):
+        return self.get_db().get_list_retention_policies()
+
+    def delete_metric_data(self, key=None, tags=None):
+        """
+        deletes a specific metric based on the key and tags
+        provided, you may also choose to delete all metrics
+        """
+        if key == 'all_metrics':
```

Comment:
let's change this so that if both `key` and `tags` are `None` this part is executed, passing `'all_metrics'` is weird.